glusterfs.spec.in

```diff
@@ -224,6 +224,10 @@
 %global __provides_exclude_from ^%{_libdir}/glusterfs/%{version}/.*$
 %endif
 
+%global bashcompdir %(pkg-config --variable=completionsdir bash-completion 2>/dev/null)
+%if "%{bashcompdir}" == ""
+%global bashcompdir ${sysconfdir}/bash_completion.d
+%endif
```

Comment:
Isn't this what configure.ac is already doing ?

Reply:
Yes, it is pretty much the same thing.

The first thing I did was to use the `@bashcompdir@` configure macro. The generated `glusterfs.spec` then contained the full path kind of hardcoded. It looked weird to me. I guess it is the idea that if someone takes the `glusterfs.spec` to build an rpm, the path should be determined dynamically at the time, and on the system, of the package build, versus at the time the source distribution was done. Does that make sense?

I saw that very recently shell completion dirs rpm macros were added in Fedora [[1]]. I guess it is the same idea but here with a dynamic pkg-config call.

[1]: https://src.fedoraproject.org/rpms/redhat-rpm-config/c/7663a47b1a48e55c17c92d08af937e340c66df86?branch=f36

Reply:
But the `glusterfs.spec` can only be generated by `configure` from `glusterfs.spec.in`, so it's already tailored for a specific environment.

Anyway I'm not an expert here and I don't know what's the best option.

Reply:
I also thought of that because the final `glusterfs.spec` (as well as the `glusterfs.spec.in) is part of the source distribution archive.

Reply:
Doing the detection (again) in the generated `glusterfs.spec` is a clean solution IMHO. In case a distribution changes the location of the bash-completion (or different RPM distributions have different locations), the same dist tarball does not need to be adapted.